### PR TITLE
add missing @action.bound

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does
Make sure `this` is always bound to the class then calling MobX action

## Which issue(s) this PR fixes
https://github.com/grafana/oncall-private/issues/2441

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: grafana-plugin/src/pages/insights/Insights.tsx [hunk 1]
```diff
@@ -72,8 +72,12 @@ const Insights = observer(() => {
         setDatasource(`${text}`);
       });
     return () => {
-      stackListener?.unsubscribe();
-      dataSourceListener?.unsubscribe();
+      if (stackListener?.unsubscribe) {
```
Comment (teodosii) [inline grafana-plugin/src/pages/insights/Insights.tsx:75]: `stackListener?.unsubscribe?.();`

File: grafana-plugin/src/pages/insights/Insights.tsx [hunk 2]
```diff
@@ -72,8 +72,12 @@ const Insights = observer(() => {
...
+        stackListener.unsubscribe();
+      }
+      if (dataSourceListener?.unsubscribe) {
+        dataSourceListener.unsubscribe();
```
Comment (teodosii) [inline grafana-plugin/src/pages/insights/Insights.tsx:79]: `dataSourceListener?.unsubscribe?.();`

Comment (teodosii) [reply grafana-plugin/src/pages/insights/Insights.tsx:79]: Up to you, but you can just reduce each of them to 1 liner each

**Review by teodosii — APPROVED** [review]

File: grafana-plugin/src/pages/insights/Insights.tsx (hunk 2 above)
Comment (brojd) [reply grafana-plugin/src/pages/insights/Insights.tsx]: That's a good suggestion, thank you, applied it:)

File: grafana-plugin/src/state/rootBaseStore/rootBaseStore.test.ts [hunk 3]
```diff
@@ -181,7 +181,7 @@ describe('rootBaseStore', () => {
     });
     isUserActionAllowed.mockReturnValueOnce(true);
     PluginState.installPlugin = jest.fn().mockResolvedValueOnce(null);
-    rootBaseStore.userStore.loadCurrentUser = mockedLoadCurrentUser;
+    Object.defineProperty(rootBaseStore.userStore, 'loadCurrentUser', { value: mockedLoadCurrentUser });
```
[the remark this answers is not in the document]
Comment (brojd) [reply grafana-plugin/src/state/rootBaseStore/rootBaseStore.test.ts:184]: @action.bound make a class method readonly and in unit tests we were directly overriding `loadCurrentUser` method which is not possible after adding `@action.bound` to it. As a replacement you can use `Object.defineProperty` which works the same but doesn't throw an error

[the remark this answers is not in the document]
Comment (teodosii) [reply grafana-plugin/src/state/rootBaseStore/rootBaseStore.test.ts:184]: Oh gotcha, so it makes it readonly, that explains it. Thanks!

**Pull request merged**